### PR TITLE
fix: prevent panic when parsing empty string

### LINKED ISSUE #748: LSP empty file crash
Opening an empty sw file causes the LSP to crash 
Screenshot 2022-02-07 at 16 38 42

### PR DESCRIPTION
**Pull request by alex-matthe**
Close #748

### CODE REVIEW

File: sway-core/src/lib.rs [hunk 1]
```diff
@@ -679,8 +679,11 @@ fn parse_root_from_pairs(
         }
     }
 
-    let fuel_ast = fuel_ast_opt.unwrap();
-    ok(fuel_ast, warnings, errors)
+    CompileResult {
+        value: fuel_ast_opt,
+        warnings,
+        errors,
+    }
```
Comment (sezna) [inline sway-core/src/lib.rs:686]: `ok()` constructs a compile result -- it is preferable to use that.
```suggestion

    ok(fuel_ast_op, warnings, errors)
```

**Review by sezna — CHANGES_REQUESTED** [review]
See comment

File: sway-core/src/lib.rs (hunk 1 above)
Comment (sezna) [reply sway-core/src/lib.rs:686]: Ah, I see. Could you add a method in `error.rs`, perhaps `pub fn new()`, that takes an option, warnings, and errors, and makes a compile result? That way we can use it in the future as well.

**Review by sezna — CHANGES_REQUESTED** [review]
Sorry for being unclear -- I meant as a _method_ of `CompileResult`. I think the name `new` is a bit too generic for a specific constructor.

To be more specific, [here](https://github.com/FuelLabs/sway/blob/master/sway-core/src/error.rs#L73), you'd want to add a block like this:
```rust
impl <T> CompileResult<T> {
    pub fn new( . . . ) etc.,
}
```

then you could use it like `CompileResult::new(...)`

**Comment by otrho** [comment]
@alex-matthe @sezna This was getting a bit stale, so I've made the changes myself.  Ready to merge.

**Review by sezna — APPROVED** [review]

**Pull request merged**